odilia/src/tower.rs

```diff
@@ -256,21 +487,22 @@ where
 		}
 		results
 	}
-	pub fn command_listener<H, T, C>(mut self, handler: H) -> Self
+	pub fn command_listener<H, T, C, R>(mut self, handler: H) -> Self
 	where
-		H: Handler<T, S, C, Response = ()> + Send + Sync + 'static,
+		H: Handler<T, S, C, Response = R> + Send + Sync + 'static,
 		C: CommandType + Send + Sync + 'static,
 		Command: TryInto<C>,
 		OdiliaError: From<<Command as TryInto<C>>::Error>,
+		R: Into<Result<(), Error>> + 'static,
 		T: 'static,
 	{
 		let tflayer: TryIntoLayer<C, Command> = TryIntoLayer::new();
 		let state = self.state.clone();
-		let ws = handler.with_state(state);
+		let ws = handler.with_state_and_fn(state, <R as Into<Result<(), Error>>>::into);
 		let tfserv = tflayer.layer(ws);
 		let dn = C::CTYPE;
 		let bs = BoxService::new(tfserv);
-		self.command_handlers.entry(dn).or_default().push(bs);
+		self.command_handlers.entry(dn).or_insert(bs);
```

Comment:
As mentioned, I'm fine with the approach that we layer the services further as to embed the commands as part of the returned Service, but I want the flexibility of command handlers such that plugins can hook into them.

Reply:
alright, but how would plugins hook in the command services themselves, since they're functions? at most, we should have separate layers for adons, which return as quickly as possible when no adons are found or none of the loaded ones are applicable to the kind of event or command present in the command vector

Reply:
TODO, obviously.